test/test-fork.c

```diff
@@ -0,0 +1,554 @@
+  if (child_pid != 0) {
+    /* parent */
+    printf("Wating on child in parent\n");
+    ASSERT(1 == read(sync_pipe[0], sync_buf, 1)); /* wait for child */
+    printf("Parent killing child\n");
+    ASSERT(0 == kill(child_pid, SIGUSR1));
+    /* run the loop, make sure we don't get the signal */
+    printf("Running loop in parent\n");
+    uv_unref((uv_handle_t*)&signal_handle); /* so the loop can exit;
+                                               we *shouldn't* get any signals */
+    run_timer_loop_once(); /* but while we share a pipe, we do, so
+                              have something active. */
+    ASSERT(0 == uv_run(uv_default_loop(), UV_RUN_ONCE));
+    printf("Signal in parent %d\n", fork_signal_cb_called);
+    ASSERT(0 == fork_signal_cb_called);
+    ASSERT(1 == write(sync_pipe2[1], "1", 1)); /* alert child */
+    printf("Waiting for child in parent\n");
+    assert_wait_child(child_pid);
+  } else {
+    /* child */
+    /* our signal handler should still be installed */
+    ASSERT(0 == uv_loop_fork(uv_default_loop()));
+    printf("Checking loop in child\n");
+    ASSERT(0 != uv_loop_alive(uv_default_loop()));
+    printf("Alerting parent in child\n");
+    ASSERT(1 == write(sync_pipe[1], "1", 1)); /* alert parent */
+    /* don't run the loop. wait for the parent to call us */
+    printf("Waiting on parent in child\n");
+    /* wait for parent. read may fail if the parent tripped an ASSERT
+       and exited, so this isn't in an ASSERT.
+    */
+    read(sync_pipe2[0], sync_buf, 1);
+    ASSERT(0 == fork_signal_cb_called);
+    printf("Exiting child \n");
+    exit(0);
+  }
+
+  MAKE_VALGRIND_HAPPY();
+  return 0;
+}
+
+static void create_file(const char* name) {
+  int r;
+  uv_file file;
+  uv_fs_t req;
+
+  r = uv_fs_open(NULL, &req, name, O_WRONLY | O_CREAT, S_IWUSR | S_IRUSR, NULL);
+  ASSERT(r >= 0);
+  file = r;
+  uv_fs_req_cleanup(&req);
+  r = uv_fs_close(NULL, &req, file, NULL);
+  ASSERT(r == 0);
+  uv_fs_req_cleanup(&req);
+}
+
+static void touch_file(const char* name) {
+  int r;
+  uv_file file;
+  uv_fs_t req;
+  uv_buf_t buf;
+
+  r = uv_fs_open(NULL, &req, name, O_RDWR, 0, NULL);
+  ASSERT(r >= 0);
+  file = r;
+  uv_fs_req_cleanup(&req);
+
+  buf = uv_buf_init("foo", 4);
+  r = uv_fs_write(NULL, &req, file, &buf, 1, -1, NULL);
+  ASSERT(r >= 0);
+  uv_fs_req_cleanup(&req);
+
+  r = uv_fs_close(NULL, &req, file, NULL);
+  ASSERT(r == 0);
+  uv_fs_req_cleanup(&req);
+}
+
+static int timer_cb_touch_called;
+
+static void timer_cb_touch(uv_timer_t* timer) {
+  uv_close((uv_handle_t*)timer, NULL);
+  touch_file("watch_file");
+  timer_cb_touch_called++;
+}
+
+static int fs_event_cb_called;
+
+static void fs_event_cb_file_current_dir(uv_fs_event_t* handle,
+                                         const char* filename, int events, int status) {
+  ASSERT(fs_event_cb_called == 0);
+  ++fs_event_cb_called;
+  ASSERT(status == 0);
+  #if defined(__APPLE__) || defined(_WIN32) || defined(__linux__)
```

Comment:
the whole test won't compile on Windows, so there is no point in having it. Even if you copied the funciton from another test, they test different things.